### PR TITLE
🐛 Fix bug when cluster is not found

### PR DESCRIPTION
**Pull request by maelk**
**What this PR does / why we need it**:
And remove the restrictions on the events triggering the reconciliation
frommetal3dataclaim to catch update events when the deletion timestamp is
set. It also improves slightly the makefile

### CODE REVIEW

File: controllers/metal3datatemplate_controller.go [hunk 1]
```diff
@@ -110,14 +107,16 @@ func (r *Metal3DataTemplateReconciler) Reconcile(req ctrl.Request) (_ ctrl.Resul
...
 
-	metadataLog = metadataLog.WithValues("cluster", cluster.Name)
-	if err := metadataMgr.SetClusterOwnerRef(cluster); err != nil {
-		return ctrl.Result{}, err
-	}
-	// Return early if the Metadata or Cluster is paused.
-	if util.IsPaused(cluster, capm3DataTemplate) {
-		metadataLog.Info("reconciliation is paused for this object")
-		return ctrl.Result{Requeue: true, RequeueAfter: requeueAfter}, nil
+	if capm3DataTemplate.Spec.ClusterName != "" && cluster.Name != "" {
```
Comment (furkatgofurov7) [inline controllers/metal3datatemplate_controller.go:110]: Why are we not checking this `cluster != nil` condition here?

Comment (maelk) [reply controllers/metal3datatemplate_controller.go:110]: because the cluster is already defined, just with empty values, so cluster would never be nil

**Pull request merged**